starwars-app/src/Components/details.jsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { useEffect, useState } from 'react';
+
+export default function CharacterProfile() {
+    const [characterDetail, setDetails] = useState();
+    const [isLoading, setIsLoading] = useState(true);
+    const [hasError, setHasError] = useState(false);
+
+    useEffect(() => {
+        fetch(`https://swapi.dev/api/people/?search={characterDetail}`)
```

Comment:
For `characterDetail`, you'll want to obtain that from the route via the `useParams` hook, then pass it in here.